### ISSUE TITLE
List of services/apps known to NOT work with Monitorr

### ISSUE THREAD

**Issue by seanvree** [Monitorr/Monitorr#89]
**BLUF**:  Apps that are known to NOT work with Monitor (PHP-CURL)r:

- DD-WRT
- NZBGet
- Mattermost

**Background**:  Monitorr uses CURL to receive HTTP status codes from services in order to determine UP/DOWN status.  This is a bit more advanced than a simple ping.  CURL actually checks to see if the webapp/site is responding whereas PING simply checks to see if there is ANY response on that port. 

**ISSUE**:  There are some services that will NOT respond to CURL requests.  We assume this is for security as to combat DDOS attacks or PING flood.  Unfortunately, there's no way around this right now other than to ask the software developer to build CURL responses into their software/UI.

You can test this by running the following command in a terminal ON THE SERVER that is hosting Monitorr. You will want to use the same exact address you use in your Monitorr/assets/config.php file.  

```
curl -I -L http://192.168.1.1
```

You'll see the response below that this particular device (in this case DDWRT) rejects the curl request.  So even tho the user can access that exact address from a browser,   Monitorr would report this service as DOWN.  

Response:  

```

C:\WINDOWS\system32>curl -I -L http://192.168.1.1
curl: (52) Empty reply from server

```

These are examples of successful CURL responses:
(The below example is to PLEX, even tho the response is "unauth" the website IS responding, therefore Monitorr would report it as UP.)
```

C:\WINDOWS\system32>curl -I -L localhost:32400
HTTP/1.1 401 Unauthorized
Content-Length: 193
Content-Type: text/html
X-Plex-Protocol: 1.0
Cache-Control: no-cache
Date: Thu, 04 Jan 2018 18:55:04 GMT


````
And below are normal successful responses:

```
C:\WINDOWS\system32>curl -I -L localhost:80
HTTP/1.1 301 Moved Permanently
Content-Length: 144
Content-Type: text/html; charset=UTF-8
Location: http://localhost/home
Server: Microsoft-IIS/10.0
 Access-Control-Allow-Origin: Access-Control-Allow-Origin: http://www.yourdomain.com
Date: Thu, 04 Jan 2018 18:55:15 GMT

HTTP/1.1 301 Moved Permanently
Content-Length: 145
Content-Type: text/html; charset=UTF-8
Location: http://localhost/home/
Server: Microsoft-IIS/10.0
 Access-Control-Allow-Origin: Access-Control-Allow-Origin: http://www.yourdomain.com
Date: Thu, 04 Jan 2018 18:55:15 GMT

HTTP/1.1 200 OK
Content-Length: 8118
Content-Type: text/html
Last-Modified: Wed, 27 Dec 2017 02:26:49 GMT
Accept-Ranges: bytes
ETag: "aea44b26ba7ed31:0"
Server: Microsoft-IIS/10.0
 Access-Control-Allow-Origin: Access-Control-Allow-Origin: http://www.yourdomain.com
Date: Thu, 04 Jan 2018 18:55:15 GMT
````

**Suggestion**:  There are a few workarounds:

1. For a service that reports as DOWN (non-reverse proxied service) despite it actually being UP - set the monitored link in config.php to a reverse proxy link - this will allow you to have a working link on your Monitorr UI given the service is actually UP. 

2.  If reverse proxy is NOT an option - you can edit the Monitorr CSS HTML code so that you don't see the ugly "RED" on your Monitorr UI and to always show an UP status. 

**Way ahead:**

We are looking into a few options such as backup simple PING, LINX, or custom response codes.   If you have any ideas. Hit us up on Discord!

NOTE:      /assets/php/check.php  ADD  the following line to NOT allow Monitorr to check beyond your reverse proxy:
```
 curl_setopt($handle, CURLOPT_FOLLOWLOCATION, FALSE);
```

**Comment by seanvree** [comment]
NOTE:      /assets/php/check.php  ADD   curl_setopt($handle, CURLOPT_FOLLOWLOCATION, true); to send request thru proxy server.

**Comment by seanvree** [comment]
delete

**Comment by seanvree** [comment]
Developing fallback method here:  https://github.com/Monitorr/Monitorr/issues/101

**Comment by seanvree** [comment]
Fix pushed to DEVELOP branch.

**Comment by seanvree** [comment]
fixed with version 0.12.7
